src/core/abr/pending_requests_store.ts

```diff
@@ -80,18 +41,18 @@ export default class PendingRequestsStore {
    * @param {string} id
    * @param {Object} payload
    */
-  public add(payload : IBeginRequestValue) : void {
-    const { id, segment, requestTimestamp } = payload;
-    this._currentRequests[id] = { segment,
-                                  requestTimestamp,
-                                  progress: [] };
+  public add(payload : IPendingRequestStoreBegin) : void {
+    const { id, requestTimestamp, content } = payload;
+    this._currentRequests[id] = { requestTimestamp,
+                                  progress: [],
+                                  content };
```

Comment:
I replaced what was just `segment` by `content`, which indicates the full context (period + adaptation + representation + segment). This is because the new `GuessingBasedChooser` now needs to filter request based on the corresponding Representation